Seagate/cortx-ha
# Eos 26559 added test file to test parsing k8s event and checking if machine-id is getting populated

# Problem Statement
- Need to add a test file to test parsing k8s event and checking if machine-id is getting populated

# Design
-  added test file: cortx-ha\ha\test\integration\k8s_monitor\test_k8s_pod_event_parsing.py

# Coding

# Test Scenario:
1. run the test file.
2. then delete some data pod
3. then you will see alrets are getting printed on stdout having proper machine-id.
4. see below output of test:

> sh-4.2# python3 test_k8s_pod_event_parsing.py
******** Testing K8s Event Parsing ********
Producer id: mock_producer, message_type: mock_message, partition: 1
Test call: mocking publish event.
{'_resource_type': 'node', '_resource_name': '3595344c9cc7469e8055d453e4fb738d', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': 'ssc-vm-g3-rhev4-2374.colo.seagate.com', '_is_status': True, '_timestamp': '1639488287'}
Test call: mocking publish event.
{'_resource_type': 'node', '_resource_name': '2920c806b724430097d2108e3ae67975', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': 'ssc-vm-g3-rhev4-2403.colo.seagate.com', '_is_status': True, '_timestamp': '1639488288'}
Test call: mocking publish event.
{'_resource_type': 'node', '_resource_name': '19a7dfc86d424a5bb3c2bc22a19b236b', '_event_type': 'online', '_k8s_container': None, '_pod': None, '_node': 'ssc-vm-g3-rhev4-2370.colo.seagate.com', '_is_status': True, '_timestamp': '1639488288'}


# Testing 
- [x] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Review Checklist 
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide
